src/pages/publication/ReferencesPanel.tsx

```diff
@@ -22,40 +29,61 @@ const StyledBox = styled.div`
 interface ReferencesPanelProps {
   goToNextTab: () => void;
   savePublication: () => void;
-  tabNumber: number;
-  selectedReferenceType: string;
 }
 
-export const ReferencesPanel: React.FC<ReferencesPanelProps> = ({
-  goToNextTab,
-  savePublication,
-  tabNumber,
-  selectedReferenceType,
-}) => {
+export const ReferencesPanel: React.FC<ReferencesPanelProps> = ({ goToNextTab, savePublication }) => {
   const { t } = useTranslation('publication');
+  const { values, setFieldTouched }: FormikProps<any> = useFormikContext();
+  const { referenceType } = values.reference;
+
+  // Validation messages won't show on fields that are not touched
+  const setAllFieldsTouched = useCallback(() => {
+    Object.values(ReferenceFieldNames).forEach(fieldName => setFieldTouched(fieldName));
+
+    switch (referenceType) {
+      case ReferenceType.BOOK:
+        Object.values(BookFieldNames).forEach(fieldName => setFieldTouched(fieldName));
+        break;
+      case ReferenceType.PUBLICATION_IN_JOURNAL:
+        Object.values(journalPublicationFieldNames).forEach(fieldName => setFieldTouched(fieldName));
+        break;
+      default:
+        break;
+    }
+  }, [setFieldTouched, referenceType]);
+
+  useEffect(() => {
+    // Set all fields as touched if user navigates away from this panel ( on unmount)
+    return () => setAllFieldsTouched();
+  }, [setAllFieldsTouched]);
+
```

Comment:
denne vil validere når man har vært innom taben, og går til en annen tab, men når man går tilbake til references, så står det ingen feilmelding. kan virke forvirrende for brukeren:

![Skjermbilde 2019-12-17 kl  11 02 21](https://user-images.githubusercontent.com/6595982/70985646-b9b7ea80-20bc-11ea-8ee6-ec3f785a02c8.png)

Reply:
ja, det er en feil på formik fields med component=Select. Feilmeldingen er der, men det mangler en eller annen kobling. I verste fall må vi gjøre det selv ved å sette opp `<FormControl`, <InputLabel>`, `<HelperText` og `<ErrorMessage>`. By default mangler label også med Select

Reply:
ok, kan evt ta det i neste PR da eller?